pkg/cloud-init/cloud-init_test.go

```diff
@@ -605,4 +610,38 @@ var _ = Describe("CloudInit", func() {
 			Expect(err).ToNot(HaveOccurred())
 		})
 	})
+
+	Context("Hostname generation", func() {
+		const cloudinitHostname = "cloudinit-hostname"
+		const userDataWithHostnameFmt = `#cloud-config
```

Comment:
Nit: *Fmt consts are usually used as printf format strings. This is more of a prefix.
Suggestion: add a comment like `#hostname: bad` to ensure comments are properly ignored.

Reply:
Thanks!
Changed `userDataWithHostnameFmt` -> `userDataWithHostnamePrefix`.
Also added a comment like you suggested